build/commands/scripts/generateCoverageReport.js

```diff
@@ -0,0 +1,131 @@
+      ].join('\n'),
+    )
+    .option('-C [build_dir]', 'build config (out/Debug, out/Release)')
+    .option('--build_type [build_type]', 'build with coverage information ')
+    .option('--target_arch [target_arch]', 'target architecture')
+    .option('--target_os <target_os>', 'target OS')
+    .option('--name [name]', 'name of the test report [defaults to coverage]', 'all')
+    .option(
+      '--clean',
+      'delete recordings and reports. Either all or those that match --tests [testSuites]',
+    )
+    .option(
+      '--tests [testSuites]',
+      'comma seperated list of testsuites to consider. By default it parses all recordings',
+    )
+    .arguments('[build_config]')
+    .action(async (buildConfig, args) => {
+      config.buildConfig = buildConfig || config.buildConfig
+      config.update(args)
+      const dirsGlob = args.tests ? `{${args.tests}}/**` : '**'
+
+      const out = `${config.outputDir}/coverage`
+      const reportPath = `${out}/report/${args.name}`
+      const profDataPath = `${reportPath}/coverage.profdata`
+      const distPath = `${reportPath}/dist`
+
+
+      const recordings = await Array.fromAsync(
+        glob(`${out}/${dirsGlob}/*.profraw`),
+      )
+
+      if (args.clean) {
+        await Promise.all(
+          recordings
+            .concat([reportPath])
+            .map((path) => rm(path, { recursive: true, force: true })),
+        )
+
+        return
+      }
+
+      const allTestSuites = [
+        ...new Set(
+          recordings
+            .map((x) => x.replace(out + '/', ''))
+            .map((x) => x.split('/'))
+            .filter((x) => x.length > 1)
+            .map((x) => x[0]),
+        ),
+      ].map((x) => `${config.outputDir}/${x}`)
+
+      // llvm-cov needs a .profdata (llvms .lcov equivalent) and
+      // binaries that include the symbols that are instrumented to generate a report
+      // Weirdly it appears that the json export is unaffected.
+      // We probably want to use gcov / genhtml instead of llvm-cov
+      // For now - if we can - only use brave_browser_tests binary
+      // since it will have the most amount of symbols to generate the report.
+      // It will still include coverage information of code paths of other test suites
+      // as long as their symbols are present in the brave_browser_test binary
+      const hasBBT = allTestSuites.find((x) =>
+        x.includes('brave_browser_tests'),
+      )
+      const testSuites = hasBBT
+        ? [`${config.outputDir}/brave_browser_tests`]
+        : allTestSuites
+
+      if (!recordings.length) {
+        console.log(`glob ${out}/${dirsGlob}/*.profraw didn't yield any recordings!\n make sure you've built with --is_coverage and ran the appropriate tests`);
+        return
+      }
+
+      process.env.CWD = config.outputDir
+      process.env.PATH = `${process.env.PATH}:${config.srcDir}/third_party/llvm-build/Release+Asserts/bin`
```

Comment:
this seems a little sketchy to rely on PATH, we should pass the absolute path to the exe instead I think

Reply:
I thought of giving preferential treatment to your local installation especially; But I actually like the idea of being more opinionated; after all we don't know if there will be some breakage given that profdata is not a really spec'ed out format like lcov